### ISSUE TITLE
NotImplementedError Using t2kdm-tests

### ISSUE THREAD

**Issue by hoganman** [t2k-software/t2kdm#22]
I encountered a NotImplementError with the t2kdm-test app when I used the lcg backend.

```bash
Configuration parameter: backend

Which backend should be used?
Supported backends: gfal, lcg

Current value: lcg
Default value: gfal

Enter new value [keep current]: 
Saving configuration in /home/mhogan/.config/t2kdm/t2kdm.conf
[mhogan@hep ~]$ t2kdm-tests
Testing ls...
Testing replicas...
Testing exists...
Traceback (most recent call last):
  File "/home/mhogan/.local/bin/t2kdm-tests", line 11, in <module>
    load_entry_point('t2kdm', 'console_scripts', 't2kdm-tests')()
  File "/home/mhogan/software/t2kdm/t2kdm/tests.py", line 226, in run_tests
    run_read_only_tests()
  File "/home/mhogan/software/t2kdm/t2kdm/tests.py", line 70, in run_read_only_tests
    assert(t2kdm.backend.exists(rep))
  File "/home/mhogan/software/t2kdm/t2kdm/cache.py", line 72, in cached_function
    return function(*args, **kwargs)
  File "/home/mhogan/software/t2kdm/t2kdm/backends.py", line 117, in exists
    return self._exists(surl, **kwargs)
  File "/home/mhogan/software/t2kdm/t2kdm/backends.py", line 112, in _exists
    raise NotImplementedError()
NotImplementedError
```
And I made sure gfal worked for me

```bash
[mhogan@hep t2kdm]$ t2kdm-tests                                                                                                                  
Testing ls...
Testing replicas...
Testing exists...
Testing checksum...
Testing state...
Testing StorageElement...
Testing TriumfStorageElement...
Testing get...
Testing check...
Testing Commands...
Testing CLI...
All done.
```

**Comment by ast0815** [comment]
Ah yes. I did not implement some of the recent functionality in the lcg backend. Since we are not supposed to be using that anyway and it is more of a fall-back solution, I figured that would be a waste of development time. I guess I should make the tests deal with this a bit more gracefully.

If someone actually needs these (generally more advanced) functions in the lcg backend, we can still put those in later.
